secure/key-manager/src/lib.rs

```diff
@@ -253,9 +259,15 @@ where
 
     pub fn perform_action(&mut self, action: Action) -> Result<(), Error> {
         match action {
-            Action::FullKeyRotation => self.rotate_consensus_key().map(|_| ()),
-            Action::SubmitKeyRotationTransaction => self.resubmit_consensus_key_transaction(),
-            Action::NoAction => Ok(()),
+            Action::FullKeyRotation => {
+                COUNTERS.full_consensus_rotations.inc();
+                self.rotate_consensus_key().map(|_| ())
+            }
+            Action::SubmitKeyRotationTransaction => {
+                COUNTERS.rotation_transaction_resubmissions.inc();
+                self.resubmit_consensus_key_transaction()
+            }
+            _ => Err(Error::InvalidActionToPerform(format!("{:?}", action))),
```

Comment:
why would we ever get here?

Reply:
This was already in the existing code. If someone called perform_action with a NoAction type, this method would just return Ok(()), which seems a little dangerous (silent failures). I've updated this to return an explicit error. Even though no one calls this method in practice with an invalid action, I felt it safer to prevent this.

Reply:
I don't follow, if it is no action and we call perform action, it should just pass through or we should eliminate no action.

Reply:
Sure! I just don't like the idea of passing an action to a function, which is then ignored. But taking a step back, I don't feel that strongly about this stuff to let it be blocking 😄I can revert!

Reply:
I see, I should have called this out in the previous PR, let's move the action handler for no action out of the execute_once, it should actually be in a handler. Hence my confusion on what was going here.

Reply:
Done, thanks!